android/jni/com/mapswithme/core/jni_helper.hpp

```diff
@@ -51,20 +53,28 @@ jobject GetNewParcelablePointD(JNIEnv * env, m2::PointD const & point);
 jobject GetNewPoint(JNIEnv * env, m2::PointD const & point);
 jobject GetNewPoint(JNIEnv * env, m2::PointI const & point);
 
-template<typename TValue, typename TToJavaFn>
-jobjectArray ToJavaArray(JNIEnv * env, jclass clazz, vector<TValue> const & src, TToJavaFn && toJavaFn)
+template<typename TIt, typename TToJavaFn>
+jobjectArray ToJavaArray(JNIEnv * env, jclass clazz, TIt begin, TIt end, size_t const size, TToJavaFn && toJavaFn)
```

Comment:
Do you need to mark as const size in the following code:
```
void foo() {
  ...
  size_t const poolSize = ...;
  ... 
}
```

Function argument is just a local variable, so why to trait them differently?
Function declaration is an excuse case.